app/controllers/branches_controller.rb

```diff
@@ -124,4 +125,16 @@ def load_repository_and_branch
     @repository || load_repository
     @branch ||= @repository.branches.where(name: params[:id]).first!
   end
+
+  def load_build_stats
+    @first_built_date = @branch.builds.first.created_at
+    @days_since_first_build = (Date.today - @first_built_date.to_date).to_i
+
+    @total_build_count = @branch.builds.count
+    @total_failure_count = @total_build_count - @branch.builds.where(:state => 'succeeded').count
```

Comment:
Why do it like that instead of

    @total_failure_count = @branch.builds.where.not(state: 'succeeded').count

[I don't think it's any faster?]